hr_course/models/hr_course.py

```diff
@@ -180,6 +183,41 @@ def cancel_course(self):
         for record in self:
             record.write(record._cancel_course_values())
 
+    @api.model
+    def send_course_notification_email(self, partner_emails):
+        email_list = partner_emails.mapped("email")
+        company_id = self.env.context.get("company_id") or self.env.company.id
+        if email_list:
+            email_template = self.env.ref(
+                "hr_course.mail_template_validity_reminder_en"
+            )
+            for email in email_list:
+                options = {
+                    "email_to": email.strip(),
+                    "email_from": self.env["res.company"]
+                    .browse(company_id)
+                    .email.strip(),
```

Comment:
```suggestion
                    "email_from": company.email.strip(),
```
browse is high costly operation try to avoid in the loops